Issue with interp

This does not compile:
```
import numpy as np

#pythran export foo(float[], float)
def foo(x, factor):
    N = len(x)
    # return np.interp(np.array(np.arange(0, N - 1, factor)), np.array(np.arange(N)), x, None, None, None) # <- This does.
    return np.interp(np.arange(0, N - 1, factor), np.arange(N), x, None, None, None)
```
The error is:
```
/Users/jlaroche/Library/Python/3.7/lib/python/site-packages/pythran/pythonic/numpy/interp.hpp:59:9: error: no viable overloaded '='
      x = x_rem;
      ~ ^ ~~~~~
/Users/jlaroche/Library/Python/3.7/lib/python/site-packages/pythran/pythonic/numpy/interp.hpp:83:12: note: in instantiation of function template specialization
      '(anonymous namespace)::pythonic::numpy::interp<(anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::pos, (anonymous
      namespace)::pythonic::numpy::details::arange_index<double> >, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::pos, (anonymous namespace)::pythonic::numpy::details::arange_index<long long> >,
      (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> >, double, double, double>' requested here
    return interp(x, xp, fp, _left, _right, 0.);
```
This is probably on my plate since I wrote the interp function. Serge, if the error tells you anything right off the bat, this will help me a lot (I'm having a hard time understanding which types are incompatible for the = operator)

--- a/pythran/pythonic/numpy/interp.hpp
+++ b/pythran/pythonic/numpy/interp.hpp
@@ -33,6 +33,8 @@ namespace numpy
     assert(std::get<0>(xp.shape()) == std::get<0>(fp.shape()));
     double outVal(0);
 
+    types::ndarray<double, types::pshape<long>> out = {x.shape(), outVal};
+
     if (period) {
       auto x_rem = pythonic::numpy::functor::remainder{}(x, period);
       auto xp_rem = pythonic::numpy::functor::remainder{}(xp, period);
@@ -56,18 +58,14 @@ namespace numpy
       auto new_fp = pythonic::numpy::functor::concatenate{}(
           pythonic::types::make_tuple(left_pad_fp, fp_sorted, right_pad_fp));
 
-      x = x_rem;
-      xp = new_xp;
-      fp = new_fp;
-      left = 0.;  // Careful: using 0 (int) would cause problems!
-      right = 0.; // Careful: using 0 (int) would cause problems!
+      auto lenxp = new_xp.size();
+      auto lenx = x_rem.size();
+      do_interp(x_rem, new_xp, new_fp, out, lenxp, lenx, 0., 0.);
+    } else {
+      auto lenxp = xp.size();
+      auto lenx = x.size();
+      do_interp(x, xp, fp, out, lenxp, lenx, left, right);
     }
-
-    types::ndarray<double, types::pshape<long>> out = {x.shape(), outVal};
-    auto lenxp = xp.size();
-    auto lenx = x.size();
-
-    do_interp(x, xp, fp, out, lenxp, lenx, left, right);
 
     return out;
   }

--- a/pythran/tests/test_numpy_func3.py
+++ b/pythran/tests/test_numpy_func3.py
@@ -547,6 +547,12 @@ def np_trim_zeros2(x):
                       numpy.random.randn(1000),
                       interp4=[NDArray[float,:],NDArray[float,:],NDArray[float,:]])
 
+    def test_interp_5(self):
+        self.run_test('def interp5(x,factor): N = len(x); import numpy as np; return np.interp(np.arange(0, N - 1, factor), np.arange(N), x, None, None, None)',
+                      numpy.random.randn(100),
+                      10.,
+                      interp5=[NDArray[float,:],float])
+
     def test_setdiff1d0(self):
         self.run_test('def setdiff1d0(x,y): import numpy as np; return np.setdiff1d(x,y)',
                       numpy.random.randn(100),

